static/src/javascripts/projects/common/modules/commercial/membership-messages.js

```diff
@@ -0,0 +1,63 @@
+define([
+    'common/utils/config',
+    'common/utils/detect',
+    'common/utils/storage',
+    'common/utils/template',
+    'common/modules/identity/api',
+    'common/modules/ui/message',
+    'text!common/views/membership-message.html',
+    'common/views/svgs'
+], function (
+    config,
+    detect,
+    storage,
+    template,
+    idApi,
+    Message,
+    messageTemplate,
+    svgs
+) {
+
+    function canRun() {
+        /**
+         * - Exclude adblock users to avoid conflicts with similar adblock Supporter message
+         * - Exclude mobile/small-screen devices
+         * - Only show for UK edition
+         * - Only show on Article pages
+         * - Only show to visitors who have viewed more than 10 pages.
+         */
+        var alreadyVisited = storage.local.get('gu.alreadyVisited') || 0;
+        return config.switches.membershipMessages &&
```

Comment:
Maybe we should start using the functionality from the https://github.com/guardian/frontend/pull/10653 here?